server/helpers/adapters_tracker.go

```diff
@@ -179,3 +213,91 @@ func (a *AdaptersTracker) UndeployAdapter(ctx context.Context, adapter models.Ad
 	a.RemoveAdapter(ctx, adapter)
 	return nil
 }
+
+func int32Ptr(n int32) *int32 {
+	return &n
+}
+
+// SetOverrideValues returns the value overrides to install/upgrade helm chart
+func (a *AdaptersTracker) setOverrideValues(adapterName string, exposedPort int32, enable bool) map[string]interface{} {
+	// Initialize the valueOverrides with default values and set config for selected adapter
+	valueOverrides := map[string]interface{}{}
+	for _, availableAdapter := range models.ListAvailableAdapters {
+		valueOverrides[availableAdapter.Name] = map[string]interface{}{
+			"enabled": false,
+		}
+		if enable && availableAdapter.Name == adapterName {
+			valueOverrides[availableAdapter.Name] = map[string]interface{}{
+				"enabled": true,
+				"service": map[string]interface{}{
+					"name":      adapterName,
+					"namespace": core.MesheryNamespace,
+					"spec": map[string]interface{}{
+						"selector": map[string]string{
+							"app": adapterName,
+						},
+						"ports": []map[string]interface{}{
+							{
+								"name":     "gRPC",
+								"protocol": "TCP",
+								"port":     exposedPort,
+							},
+						},
+						"type": "ClusterIP",
+					},
+				},
+			}
+		}
+	}
+
+	// Enable all already existing adapters
+	for _, setAdapter := range a.adapters {
+		setAdapterPort, _ := strconv.Atoi(strings.Split(setAdapter.Location, ":")[1])
+		valueOverrides[setAdapter.Name] = map[string]interface{}{
+			"enabled": true,
+			"service": setService(setAdapter.Name, setAdapterPort),
+		}
+	}
+
+	return valueOverrides
+}
+
+// Apply Helm charts for adapter deployment and service
+func (a *AdaptersTracker) applyHelmCharts(kubeClient *meshkitkube.Client, adapterName string, exposedPort int32, enable bool) error {
+	// get value overrides to install the helm chart
+	overrideValues := a.setOverrideValues(adapterName, exposedPort, enable)
+
+	// install the helm charts with specified override values
+	return kubeClient.ApplyHelmChart(meshkitkube.ApplyHelmChartConfig{
+		Namespace:       core.MesheryNamespace,
+		ReleaseName:     "meshery",
+		CreateNamespace: true,
+		ChartLocation: meshkitkube.HelmChartLocation{
+			Repository: utils.HelmChartURL,
+			Chart:      utils.HelmChartName,
+		},
+		Action:         meshkitkube.INSTALL,
+		DryRun:         false,
+		OverrideValues: overrideValues,
+	})
+}
+
+func setService(name string, port int) map[string]interface{} {
```

Comment:
Why can't we use the Service struct from k8s.io/api for this? Instead of manually creating map[string]interface{}